### DISCUSSION TITLE
Different behaviour in V7 compared with V6

### DISCUSSION THREAD

**Discussion by MonalisaBaltatescu** [react-hook-form/react-hook-form#8800] [General] [answered]
Hi! During my upgrade to version 7 (v7.33.1) from version 6 (v6.2.0) I've noticed a difference between what `render` callback from `<Controller />` element.

[Version 6 example](https://codesandbox.io/s/react-hook-form-6-playground-9k5z3y)
[Version 7 example](https://codesandbox.io/s/react-hook-form-7-playground-4ufv9q)

To be more specific, here is my use case:
- I have a form which dynamically fills in with fields (field names come from an api)
- For a checkbox field, I have a custom element accepting a `ref` which I render inside a `<Controller />`
- I don't set default values because I don't know exactly what fields do I have in the form at the moment I define it with `useForm`

1. Version 6 returned the input value as **false**
2. Version 7 returns the input value as **undefined** (that triggers validation for the input on the first render which is not what I want)

To spot the difference, see in the console the inputs for the two versions. 

**My question is if this behaviour is intended for version 7 and how can we overcome the issue if so?** 

Currently I use `resetField` to set a default value on `useEffect`.

**Another question would be why in V7 `ref` returned from render is a function now compared with V6 where it was an object? Has this something to do with my issue?**

**Comment by bluebill1049** [comment, marked as answer]
you probably should provide a default value for this case. v7 derivate state value from the defaultValues and no longer assume fallback to HTML (input) alike value

**Reply by MonalisaBaltatescu** [reply]
Got it! 

However, is there any other way to set default values at a later point in time than when defining the form with `useForm`?

**Reply by bluebill1049** [reply]
you may have to use `reset`

**Reply by MonalisaBaltatescu** [reply]
Thank you!
